src/main/java/edu/harvard/iq/dataverse/engine/command/impl/UpdateDataverseAttributeCommand.java

```diff
@@ -0,0 +1,110 @@
+        this.attributeName = attributeName;
+        this.attributeValue = attributeValue;
+    }
+
+    @Override
+    public Dataverse execute(CommandContext ctxt) throws CommandException {
+        switch (attributeName) {
+            case ATTRIBUTE_ALIAS:
+            case ATTRIBUTE_NAME:
+            case ATTRIBUTE_DESCRIPTION:
+            case ATTRIBUTE_AFFILIATION:
+                setStringAttribute(attributeName, attributeValue);
+                break;
+            case ATTRIBUTE_FILE_PIDS_ENABLED:
+                setBooleanAttributeForFilePIDs(ctxt);
+                break;
+            default:
+                throw new IllegalCommandException("'" + attributeName + "' is not a supported attribute", this);
+        }
+
+        return ctxt.engine().submit(new UpdateDataverseCommand(dataverse, null, null, getRequest(), null));
+    }
+
+    /**
+     * Helper method to set a string attribute.
+     *
+     * @param attributeName  The name of the attribute.
+     * @param attributeValue The value of the attribute (must be a String).
+     * @throws IllegalCommandException if the provided attribute value is not of String type.
+     */
+    private void setStringAttribute(String attributeName, Object attributeValue) throws IllegalCommandException {
+        if (!(attributeValue instanceof String stringValue)) {
+            throw new IllegalCommandException("'" + attributeName + "' requires a string value", this);
+        }
+
+        switch (attributeName) {
+            case ATTRIBUTE_ALIAS:
+                dataverse.setAlias(stringValue);
+                break;
+            case ATTRIBUTE_NAME:
+                dataverse.setName(stringValue);
+                break;
+            case ATTRIBUTE_DESCRIPTION:
+                dataverse.setDescription(stringValue);
+                break;
+            case ATTRIBUTE_AFFILIATION:
+                dataverse.setAffiliation(stringValue);
+                break;
+            default:
+                throw new IllegalCommandException("Unsupported string attribute: " + attributeName, this);
+        }
+    }
+
+    /**
+     * Helper method to handle the "filePIDsEnabled" boolean attribute.
+     *
+     * @param ctxt The command context.
+     * @throws PermissionException if the user doesn't have permission to modify this attribute.
+     */
+    private void setBooleanAttributeForFilePIDs(CommandContext ctxt) throws CommandException {
+        if (!getRequest().getUser().isSuperuser()) {
+            throw new PermissionException("You must be a superuser to change this setting",
+                    this, Collections.singleton(Permission.EditDataset), dataverse);
+        }
+        if (!ctxt.settings().isTrueForKey(SettingsServiceBean.Key.AllowEnablingFilePIDsPerCollection, false)) {
+            throw new PermissionException("Changing File PID policy per collection is not enabled on this server",
+                    this, Collections.singleton(Permission.EditDataset), dataverse);
```

Comment:
Minor point, but if we get here we have a super user so wouldn't it be an illegal command exception - not a permission exception?

Reply:
I initially thought the same, but I chose to use the PermissionException since that was the original exception thrown before moving this logic into the command. It's not a significant backward compatibility issue, but that was the reason behind my decision. However, I can switch to the IllegalCommandException if you think it fits better.